cmd/cortex/main.go

```diff
@@ -184,6 +195,12 @@ func LoadConfig(filename string, expandENV bool, cfg *cortex.Config) error {
 		return errors.Wrap(err, "Error reading config file")
 	}
 
+	// create a sha256 hash of the config before expansion and expose it via
+	// the config_info metric
+	hash := sha256.Sum256(buf)
+	configInfo.Reset()
+	configInfo.WithLabelValues(fmt.Sprintf("sha256:%x", hash)).Set(1)
```

Comment:
Just took a look at how Alertmanager is doing this, and I think we should follow their approach. 

md5 sum of config loaded from disk, exposed as the metric _value itself_ should work better. Ref: https://github.com/prometheus/alertmanager/pull/751

Will be easier to build dashboards as well as alerts.

Reply:
I agree it would be easier to build alert, I don't see us building alerts and dashboards for this.

That only works for short checksums (I'd argue it fails for MD5 which is 16 bytes and thus requires 2 float64 values, and even may lead to values like NaN, +Inf). I don't think that is a good pattern, and would stay with the current design.

Reply:
Even then, building alert is not difficult... eg. `count(count(cortex_config_info) by (hash))` gives you number of different config files. If it's > 1, you can alert.

Reply:
Thanks for explaining @pstibrany; this makes perfect sense. :)

One last thing I'd like to add regarding dashboards, I envision a dashboard for monitoring a new config rollout to a multi-node cluster running the single binary - there it would be easy to track which nodes have picked up the new config if we plot the hash itself (it can be confusing to check labels to monitor the rollout). What would you recommend for that?

Reply:
Would plotting `count(cortex_config_info) by (hash)` work? One hash would go down to 0 over time (and then disappear), another one up.

Reply:
Yes it think so, that should do what we need